.github/workflows/publish.yaml

```diff
@@ -6,19 +6,22 @@ on:
       master
 jobs:
   build_deploy:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - name: Set up JDK 11
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
       with:
+        distribution: 'adopt'
         java-version: 11
     - name: Set up node
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v3
       with:
         node-version: 12.21.0
```

Comment:
can we also safely update to node 16?

Reply:
I built locally using node 16  (16.14.0) and it worked.